### PR TITLE
Place::Local(x) is now Place::Base(PlaceBase::Local(x))

### PR DESCRIPTION
**Pull request by spastorino**
We need to merge this after the beta cut for this https://github.com/rust-lang/rust/pull/58631 to work.
/cc @oli-obk

### CODE REVIEW

**Comment by spastorino** [comment]
@oli-obk this is mergeable now, just force pushed again because there was a conflict.

**Comment by mati865** [comment]
@spastorino it doesn't build:
```
error: unexpected end of macro invocation
 --> <::if_chain::__if_chain macros>:6:46
  |
6 | __if_chain ! { @ init ( $ ( $ tt ) * $ head ) $ ( $ tail ) * } } ; (
  |                                              ^ missing tokens in macro arguments
warning: unused imports: `is_copy`, `walk_ptrs_ty_depth`
 --> clippy_lints/src/redundant_clone.rs:2:25
  |
2 |     has_drop, in_macro, is_copy, match_def_path, match_type, paths, snippet_opt, span_lint_node,
  |                         ^^^^^^^
3 |     span_lint_node_and_then, walk_ptrs_ty_depth,
  |                              ^^^^^^^^^^^^^^^^^^
  |
  = note: #[warn(unused_imports)] on by default
warning: unused import: `TerminatorKind`
  --> clippy_lints/src/redundant_clone.rs:13:5
   |
13 |     TerminatorKind,
   |     ^^^^^^^^^^^^^^
error: aborting due to previous error
error: Could not compile `clippy_lints`.
```

https://travis-ci.com/rust-lang/rust-clippy/jobs/181611795

File: clippy_lints/src/redundant_clone.rs [hunk 1]
```diff
@@ -229,7 +229,7 @@ fn is_call_with_ref_arg<'tcx>(
     if_chain! {
         if let TerminatorKind::Call { func, args, destination, .. } = kind;
         if args.len() == 1;
-        if let mir::Operand::Move(mir::Place::Local(local)) = &args[0];
+        if let mir::Operand::Move(mir::Place::Base(PlaceBase(Local(local))) = &args[0];
```
Comment (mati865) [inline clippy_lints/src/redundant_clone.rs]:
```suggestion
        if let mir::Operand::Move(mir::Place::Base(PlaceBase(Local(local)))) = &args[0];
```
It should fix the error.

Comment (spastorino) [reply clippy_lints/src/redundant_clone.rs]: @mati865 doh but that's not the issue. `PlaceBase::Local` is the thing

Comment (mati865) [reply clippy_lints/src/redundant_clone.rs]: I admit I haven't checked it deeply, thanks for the explanation.

**Pull request merged**